pkg/datastoresrp/setup/setup.go

```diff
@@ -0,0 +1,167 @@
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateMongoDatabaseTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Delete: builder.Operation[datamodel.MongoDatabase]{
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewDeleteResource[*datamodel.MongoDatabase, datamodel.MongoDatabase](options, &mongo_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateRedisCacheTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Custom: map[string]builder.Operation[datamodel.MongoDatabase]{
+			"listsecrets": {
+				APIController: mongo_ctrl.NewListSecretsMongoDatabase,
+			},
+		},
+	})
+
+	_ = ns.AddResource("sqlDatabases", &builder.ResourceOption[*datamodel.SqlDatabase, datamodel.SqlDatabase]{
+		RequestConverter:  converter.SqlDatabaseDataModelFromVersioned,
+		ResponseConverter: converter.SqlDatabaseDataModelToVersioned,
+
+		Put: builder.Operation[datamodel.SqlDatabase]{
+			UpdateFilters: []apictrl.UpdateFilter[datamodel.SqlDatabase]{
+				rp_frontend.PrepareRadiusResource[*datamodel.SqlDatabase],
+			},
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewCreateOrUpdateResource[*datamodel.SqlDatabase, datamodel.SqlDatabase](options, &sql_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ResourceClient, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateMongoDatabaseTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Patch: builder.Operation[datamodel.SqlDatabase]{
+			UpdateFilters: []apictrl.UpdateFilter[datamodel.SqlDatabase]{
+				rp_frontend.PrepareRadiusResource[*datamodel.SqlDatabase],
+			},
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewCreateOrUpdateResource[*datamodel.SqlDatabase, datamodel.SqlDatabase](options, &sql_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ResourceClient, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateMongoDatabaseTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Delete: builder.Operation[datamodel.SqlDatabase]{
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewDeleteResource[*datamodel.SqlDatabase, datamodel.SqlDatabase](options, &sql_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateRedisCacheTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Custom: map[string]builder.Operation[datamodel.SqlDatabase]{
+			"listsecrets": {
+				APIController: sql_ctrl.NewListSecretsSqlDatabase,
+			},
+		},
+	})
+
+	// Optional
```

Comment:
Why is this optional?

Reply:
This shouldn't be optional actually so going to remove. It was carried over from the corerp setup but if we don't register the operations, it won't work

Reply:
I actually don't think this is optional but this is copied over from the corerp setup so going to keep it for now. Will sync with @youngbupark about this